src/lib/utils/loadstor.h

```diff
@@ -82,6 +83,52 @@ inline constexpr uint64_t make_uint64(
            (static_cast<uint64_t>(i6) << 8) | (static_cast<uint64_t>(i7)));
 }
 
+/**
+* Load a big-endian unsigned integer
+* @param in a fixed-length span with some bytes
+* @return T loaded from in, as a big-endian value
+*/
+template <concepts::unsigned_integral T>
+inline constexpr T load_be(std::span<const uint8_t, sizeof(T)> in) {
+   if constexpr(sizeof(T) == 1) {
+      return static_cast<T>(in[0]);
+   } else {
+#if defined(BOTAN_TARGET_CPU_IS_BIG_ENDIAN)
+      return typecast_copy<T>(in);
+#elif defined(BOTAN_TARGET_CPU_IS_LITTLE_ENDIAN)
+      return reverse_bytes(typecast_copy<T>(in));
+#else
+      return [&]<size_t... i>(std::index_sequence<i...>) {
+         return ((static_cast<T>(in[i]) << ((sizeof(T) - i - 1) * 8)) | ...);
+      }
+      (std::make_index_sequence<sizeof(T)>());
```

Comment:
What kind of code does this generate on CPUs where the endian is unknown at compile time? You can test this by commenting out `endian little` in `build-data/arch/x86_64.txt`

Reply:
Trying [on godbolt, with clang 12 x86_64](https://godbolt.org/z/aexr9G5rT), it boils it down to a `bswap` (for `load_be`) `mov` (for `load_le`). Same with [GCC 11.3](https://godbolt.org/z/Ev8h6fE7E). [MSVC v19](https://godbolt.org/z/jYYonqqdY) seems to make it into an unrolled loop at least.

Reply:
OK looks fine. MSVC codegen is always terrible anyway *and* the platforms where endianness is uncertain are typically obscure bi-endian systems like MIPS or SH where GCC/Clang are the dominant compilers.

(Occurs to me that we could start using `cpp`-based macro sniffing as we do already to check the compiler version which could probably eliminate this uncertain-endian state entirely. No need to address that here unless you want to.)

Reply:
> No need to address that here unless you want to.

I think I'll pass. 😏 I'd rather have another look at `copy_out_vec_be`, but maybe in a follow-up.

> MSVC codegen is always terrible anyway

I was actually quite amazed that GCC/clang figured out that we're just `bswap`-ing here. I did expect an unrolled loop at most.